core/controller/cluster/controller_basic.go

```diff
@@ -1090,3 +1105,33 @@ func validateClusterName(name string) error {
 func isClusterStatusUnstable(status string) bool {
 	return status == common.ClusterStatusCreating || status == common.ClusterStatusDeleting
 }
+
+var (
```

Comment:
Why are these two variables put outside of the func addIsFavoriteForClusters?

Reply:
no need to allocate memory for these two variables every execute this method

Reply:
no need to allocate memory for these two variables every execute this method

Reply:
no need to allocate memory for these two variables every execute this method

Reply:
Is it possible that concurrency problem could happen?

Reply:
it's readonly variable